src/components/UserSelector.tsx

```diff
@@ -1,23 +1,24 @@
-import React, { useContext, useEffect, useState } from 'react';
+import { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
-import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { actions as userActions } from '../features/usersSlice';
+import { actions as authorAction } from '../features/authorSlice';
+import { getUsers } from '../api/users';
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
-
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
+export const UserSelector = () => {
   // `users` are loaded from the API, so for the performance reasons
   // we load them once in the `UsersContext` when the `App` is opened
   // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+
+  const dispatch = useAppDispatch();
+  const { users } = useAppSelector(state => state.users);
+  const { author } = useAppSelector(state => state.author);
+
+  useEffect(() => {
```

Comment:
You should add a dependency array with `dispatch` to the `useEffect` hook, like this: `useEffect(() => { ... }, [dispatch]);` This ensures the effect only runs when the `dispatch` function changes.